src/extension/helper_process.c

```diff
@@ -219,13 +221,22 @@ bool dd_on_runtime_path_update(zval *nullable old_val, zval *nonnull new_val)
 {
     UNUSED(old_val);
 
+    uid_t uid = getuid();
+    gid_t gid = getgid();
+
+    size_t sock_name_len = snprintf(NULL, 0, DD_SOCK_PATH_RE, uid, gid) + 1;
+    char sock_name[sock_name_len];
+    snprintf(sock_name, sock_name_len, DD_SOCK_PATH_RE, uid, gid);
     pefree(_mgr.socket_path, 1);
     _mgr.socket_path = _concat_paths(
-        Z_STRVAL_P(new_val), Z_STRLEN_P(new_val), ZEND_STRL(DD_SOCKET_PATH));
+        Z_STRVAL_P(new_val), Z_STRLEN_P(new_val), sock_name, sock_name_len);
```

Comment:
I don't think this makes sense, why can't you include the `runtime_path` on the snprintf and avoid calling `_concat_paths` altogether?